server/models/image_item.py

```diff
@@ -0,0 +1,242 @@
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import collections
+import os
+
+from girder.api.rest import RestException
+from girder.models.model_base import ValidationException
+from girder.models.item import Item
+from girder.plugins.worker import utils as workerUtils
+from girder.plugins.jobs.constants import JobStatus
+
+from ..tilesource import TestTileSource, TiffGirderTileSource, \
+    SVSGirderTileSource, TileSourceException, TileSourceAssetstoreException
+
+
+class ImageItem(Item):
+    # We try these sources in this order.  The first entry is the fallback for
+    # items that antedate there being multiple options.
+    AvailableSources = collections.OrderedDict([
+        ('tiff', TiffGirderTileSource),
+        ('svs', SVSGirderTileSource),
+    ])
+
+    def initialize(self):
+        super(ImageItem, self).initialize()
+
+    def createImageItem(self, item, fileObj, user=None, token=None):
+        if 'largeImage' in item:
+            # TODO: automatically delete the existing large file
+            # TODO: this should raise a GirderException, but still return 400
+            raise RestException('Item already has a largeImage set.')
+        if fileObj['itemId'] != item['_id']:
+            raise RestException('The provided file must be in the provided '
+                                'item.')
+
+        if item.get('expectedLargeImage'):
+            del item['expectedLargeImage']
+        if item.get('largeImageSourceName'):
+            del item['largeImageSourceName']
+
+        item['largeImage'] = fileObj['_id']
+        job = None
+        for sourceName in self.AvailableSources:
+            try:
+                self.AvailableSources[sourceName](item)
+                item['largeImageSourceName'] = sourceName
+                break
+            except TileSourceAssetstoreException:
+                raise
+            except TileSourceException:
+                continue  # We want to try the next source
+        else:
+            # No source was successful
+            del item['largeImage']
+            job = self._createLargeImageJob(item, fileObj, user, token)
+            item['expectedLargeImage'] = True
+            item['largeImageOriginalId'] = fileObj['_id']
+            item['largeImageJobId'] = job['_id']
+
+        self.save(item)
+        return job
+
+    def _createLargeImageJob(self, item, fileObj, user, token):
+        path = os.path.join(os.path.dirname(__file__), 'create_tiff.py')
```

Comment:
This path is no longer correct.  create_tiff.py is in the parent directory, so you either need to move it or change this to `path = os.path.join(os.path.dirname(__file__), '..', 'create_tiff.py')`